### PR TITLE
Remove the Extensions categories, page restyle

### PR DESCRIPTION
**Pull request by mfrecchiami**
This PR restyles the extensions' filtering.
The new layout is based on [this](https://www.figma.com/proto/35BB9OTPKJkNU1bVYXhkKc/solidus.io?page-id=2162%3A142&node-id=2284%3A222&viewport=1909%2C144%2C0.8847187161445618&scaling=min-zoom) but we did some improvements working on this PR.

### What is changed
We've preferred to show all the extensions divided by their category. So _Official_, _Community_, _Third Part_ don't exist anymore.
Screenshot 2021-06-17 at 15 29 39

Screenshot 2021-06-17 at 15 30 29

_Official_ is changed in _Certified_

### Mobile view
Instead of hiding every extension card into an accordion, now the extensions are visible by scrolling horizontally.
[**Here's**](https://www.loom.com/share/2758dbc3700547fd9c3ec7aff91909b0) the overview on mobile.

### CODE REVIEW

File: source/extensions.html.erb [hunk 1]
```diff
@@ -3,54 +3,67 @@ title: Extensions
...
+        </h3>
+
+        <div id="collapse-most-pupular" class="content collapse" data-parent="#accordion">
+          <div class="extensions">
+            <% data.extensions.each do |extensions, extension| %>
+                <% if extension[:most_popular] %>
+                  
+                <% end %>
+            <% end %>
+          </div>
```
Comment (mfrecchiami) [inline source/extensions.html.erb:61]: Hi @kennyadsl 👋 I need your help! I want to display the extensions which have `most_popular: true`. I've done it with the code above this comment but I cannot include the same partial we're using `_extension.html.erb`. I've found some rails conflicts about the `<% extension.categories.each do |category| %>` and `<% extension.ci.build_statuses(versions: versions).each_pair do |version, build| %>`. Maybe is there a better way to group extensions that have the `most_popular` value true? Should I create a new helper method like the following?
```
def group_popular(extensions)
    extensions.group_by(&:most_popular).sort_by { |group_name, _| group_name }.to_h
  end
```
(I've tried also this way, but there's always some errors like the ones I mentioned in this comment 😭 )

Comment (mfrecchiami) [reply source/extensions.html.erb:61]: The big picture is: we're going to show the `Most Popular` group/category as the first of the list, only when a user doesn't filter the extensions. So the single extension's card shown in this category will be repeated even into their own group (for example if I set as Most Popular the PayPal extension, that will be shown into the Most Popular group and even into the Payments group).

**Comment by aldesantis** [comment]
@mfrecchiami couple of things:

- If I e.g. filter by Content Management, and then remove the filter, the list of extensions is empty. I would expect it to display all extensions again when I remove the filter.
- I think we should remove the category from the individual extension card, now that the category is in the heading.

**Comment by mfrecchiami** [comment]
> @mfrecchiami couple of things:
> 
> * If I e.g. filter by Content Management, and then remove the filter, the list of extensions is empty. I would expect it to display all extensions again when I remove the filter.
> * I think we should remove the category from the individual extension card, now that the category is in the heading.

@aldesantis thanks for your feedback! You're right, when all the filters are deselected I should return the initial status, so showing all the extension categories. For do it I need some help, because it implies to change completely the behavior of the filters. In fact in the original version we showed a message when all the filters were deselected (I wanted to do the same here, but I miss something and when all the filters are deselected the alert doesn't appear). I don't know what is the best approach to change it, could you help me with that?

The category on each cards it could be helpful just for cards that appears under the Most Popular category. Maybe we can keep it just for that and hide it when the card is shown under its category. What do you think about it?

**Comment by aldesantis** [comment]
@mfrecchiami I agree, let's only keep the category for the Most Popular extensions.

As for changing the JS, can you get some help from @davidedistefano? Not sure I'll have much time to take care of that in the next few days.

**Comment by davidedistefano** [comment]
@mfrecchiami We can take a look at it together tomorrow if you want.

**Comment by mfrecchiami** [comment]
Sounds good, thank you both guys! 👍

**Comment by mfrecchiami** [comment]
@aldesantis hi! We've done some changes by following your primary feedback. We're going to show 3 collections by default, which are:
- Most Popular (that includes the _official extensions_)
- Community
- Third-Party

By selecting some of the category filters like Payments, Auditing, Compliance, and so on, the 3 collections disappear; when all the category filters are unchecked, the page returns to the initial status showing the 3 main collections.

Let us know what do you think about it, even if the 3 main categories make sense for you. Thanks!

**Comment by mfrecchiami** [comment]
> @aldesantis hi! We've done some changes by following your primary feedback. We're going to show 3 collections by default, which are:
> 
> * Most Popular (that includes the _official extensions_)
> * Community
> * Third-Party
> 
> By selecting some of the category filters like Payments, Auditing, Compliance, and so on, the 3 collections disappear; when all the category filters are unchecked, the page returns to the initial status showing the 3 main collections.
> 
> Let us know what do you think about it, even if the 3 main categories make sense for you. Thanks!

Hi @aldesantis! I've created a second version without the 3 main categories and you can see the difference between the two versions by these short videos:

- [current version of this PR](https://www.loom.com/share/46a7d5fa81eb428ca7b95fe71477f930)
- [version with extensions categories only](https://www.loom.com/share/f861773c49b74f77b61ff9871fe0f260)

As discussed with @davidedistefano the benefit of having 3 main categories to display by default is that we allow the user to filter the extensions by selecting the extension category he/she wants to view actually - instead of displaying all the extensions and then let the user deselected the unwanted categories. Furthermore, having the Most Popular category gives an immediate overview of the most relevant extensions.
I'd prefer the first solution, but surely we should define which categories display as "Most Popular". Which version makes more sense for you?

**Comment by aldesantis** [comment]
@mfrecchiami I agree that it's annoying to have to de-select all categories, but I also don't like the three "collections", as that separation doesn't make a lot of sense, semantically.

What if we defaulted to displaying all categories when no categories are selected?

**Comment by mfrecchiami** [comment]
@aldesantis I've tried what you suggested, what do you think?

**Comment by aldesantis** [comment]
@mfrecchiami that look much better, thanks!

**Comment by mfrecchiami** [comment]
@davidedistefano what do you think about the final version? Is it good to go?

**Review by davidedistefano — APPROVED** [review]
👍

**Pull request merged**